### PR TITLE
PRC dashboard

### PR DESCRIPTION
**Pull request by paulineribeyre**
Merge the Pandemic Response Commons dashboard

### CODE REVIEW

File: runWebpack.sh [hunk 1]
```diff
@@ -167,6 +167,7 @@ if [[ "$NODE_ENV" == "dev" || "$NODE_ENV" == "auto" ]]; then
   echo ./node_modules/.bin/webpack-dev-server
   ./node_modules/.bin/webpack-dev-server
 else
+  export NODE_OPTIONS='--max-old-space-size=4096'
```
Comment (mfshao) [inline runWebpack.sh:170]: is this needed?

File: src/Covid19Dashboard/ChartCarousel/index.jsx [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+import 'slick-carousel/slick/slick.css';
+import 'slick-carousel/slick/slick-theme.css';
+
+import { covid19DashboardConfig } from '../../localconf';
+import Spinner from '../../components/Spinner';
+import PlotChart from '../PlotChart';
+import './ChartCarousel.less';
+
+let dataUrl = covid19DashboardConfig ? covid19DashboardConfig.dataUrl : '';
+dataUrl = !dataUrl.endsWith('/') ? `${dataUrl}/` : dataUrl;
```
Comment (mfshao) [inline src/Covid19Dashboard/ChartCarousel/index.jsx:13]: minor: can do this in `localconf.js`

File: src/Covid19Dashboard/IllinoisMapChart/index.jsx [hunk 3]
```diff
@@ -0,0 +1,258 @@
...
+          {...this.state.viewport}
+          {...this.state.mapSize} // after viewport to avoid size overwrite
+          onViewportChange={(viewport) => {
+            this.setState({ viewport });
+          }}
+          onHover={this.onHover}
+          onClick={this.onClick}
+          dragRotate={false}
+          touchRotate={false}
+          // maxBounds={[ // doesn't work
```
Comment (mfshao) [inline src/Covid19Dashboard/IllinoisMapChart/index.jsx:210]: remove this?

File: src/Covid19Dashboard/PlotChart/index.jsx [hunk 4]
```diff
@@ -0,0 +1,333 @@
...
+  LineChart, Line, BarChart, Bar, XAxis, YAxis, CartesianGrid, Tooltip, Legend, ResponsiveContainer,
+} from 'recharts';
+
+import Spinner from '../../components/Spinner';
+import { numberWithCommas, downloadFromGuppy } from '../dataUtils.js';
+import './PlotChart.less';
+
+
+class PlotChartAxisTick extends React.Component {
+  /* eslint-disable react/prop-types */
```
Comment (mfshao) [inline src/Covid19Dashboard/PlotChart/index.jsx:14]: why this rule is disabled in here?

File: src/Covid19Dashboard/PlotChart/index.jsx [hunk 5]
```diff
@@ -0,0 +1,333 @@
...
+    if (type === 'date') {
+      formattedValue = `${moment(formattedValue).month() + 1}/${moment(formattedValue).date()}`;
+    } else if (type === 'number') {
+      formattedValue = numberWithCommas(formattedValue);
+    }
+    // else { // type === 'string'
+    //   const maxLength = 4;
+    //   formattedValue = formattedValue.slice(0, maxLength)
+    //     + (formattedValue.length > maxLength ? '.' : '');
+    // }
```
Comment (mfshao) [inline src/Covid19Dashboard/PlotChart/index.jsx:29]: remove these?

File: src/Covid19Dashboard/index.jsx [hunk 6]
```diff
@@ -0,0 +1,397 @@
...
+    // const seirChartData = displaySeirPlot ? [
+    //   {
+    //     data: this.props.seirObservedChartData,
+    //     name: 'Observed Cases',
+    //   },
+    //   {
+    //     data: this.props.seirSimulatedChartData,
+    //     name: 'Simulated Cases',
+    //   },
+    // ] : [];
```
Comment (mfshao) [inline src/Covid19Dashboard/index.jsx:231]: do we want to keep this in here?

File: webpack.config.js [hunk 7]
```diff
@@ -141,6 +142,7 @@ module.exports = {
     hot: true,
     port: 9443,
     https: true,
+    host: '0.0.0.0',
```
Comment (mfshao) [inline webpack.config.js:145]: i assume this will make the dev server be accessible externally? do we need this

**Review by mfshao — COMMENTED** [review]
pretty solid since this thing has been running in PRC for a long while

i'm thinking probably we can leverage https://github.com/uc-cdis/data-portal/pull/721 and bundle this covid19 dashboard to a dedicated entrypoint, since it is more like a project specific thing

but that can you another ticket 😛

File: runWebpack.sh (hunk 1 above)
Comment (paulineribeyre) [reply runWebpack.sh:170]: https://github.com/uc-cdis/data-portal/pull/674#discussion_r412477816 😆

File: src/Covid19Dashboard/PlotChart/index.jsx (hunk 4 above)
Comment (paulineribeyre) [reply src/Covid19Dashboard/PlotChart/index.jsx:14]: because `PlotChartAxisTick` contains `const { x, y, payload, axis, type } = this.props;` and then `payload.value` so eslint complains the prop types are not described

`payload` comes from `XAxis.tick` and `YAxis.tick` and i'm not sure what else it contains

tldr: laziness 🤔

File: webpack.config.js (hunk 7 above)
Comment (paulineribeyre) [reply webpack.config.js:145]: i think Zac added this to be able to run portal rom a dockerfile. i'll remove it

File: runWebpack.sh (hunk 1 above)
Comment (mfshao) [reply runWebpack.sh:170]: proof of this person doesn't memorize things 🤦

File: src/Covid19Dashboard/PlotChart/index.jsx (hunk 4 above)
Comment (mfshao) [reply src/Covid19Dashboard/PlotChart/index.jsx:14]: oh i'd like we put them into the `propTypes` but i also understand the tldr part

**Comment by paulineribeyre** [comment]
> i'm thinking probably we can leverage #721 and bundle this covid19 dashboard to a dedicated entrypoint, since it is more like a project specific thing

I created ticket COV-462 for this because I don't have time to look into it right now 😅  thanks though that sounds interesting

File: src/Covid19Dashboard/PlotChart/index.jsx (hunk 5 above)
Comment (paulineribeyre) [reply src/Covid19Dashboard/PlotChart/index.jsx:29]: there's a ticket to add this back and make it configurable, so i left it in

**Review by mfshao — APPROVED** [review]
🎉
